src/main/scala/org/deepdive/ddlog/DeepDiveLogPrettyPrinter.scala

```diff
@@ -48,16 +48,61 @@ object DeepDiveLogPrettyPrinter extends DeepDiveLogHandler {
        |""".stripMargin
   }
 
+  // print an expression
+  def printExpr(e: Expr) : String = {
+    e match {
+      case VarExpr(name) => name
+      case ConstExpr(value) => {
+        if (value.startsWith("'")) s""" "${value.stripPrefix("'").stripSuffix("'")}" """
+        else value
+      }
+      case FuncExpr(function, args, agg) => {
+        val resolvedArgs = args map (x => printExpr(x))
+        s"${function}(${resolvedArgs.mkString(", ")})"
+      }
+      case BinaryOpExpr(lhs, op, rhs) => s"${printExpr(lhs)} ${op} ${printExpr(rhs)}"
+    }
+  }
+
+  // print a condition
+  def printCond(cond: Cond) : String = {
+    cond match {
+      case ComparisonCond(lhs, op, rhs) => s"${printExpr(lhs)} ${op} ${printExpr(rhs)}"
+      case NegationCond(c) => s"![${printCond(c)}]"
+      case BinaryOpCond(lhs, op, rhs) => {
+        op match {
+          case LogicOperator.AND => s"[${printCond(lhs)}], [${printCond(rhs)}]" 
+          case LogicOperator.OR  => s"[${printCond(lhs)}]; [${printCond(rhs)}]"
```

Comment:
I think the simplest bracket rule to be unambiguous is to surround any non leaf node or recursive cases. So if we just surround all cases other than the comparison I think it will place the min number of brackets, right? This won't be possible if done at each point of recursion as it's currently doing.